c10/metal/special_math.h

```diff
@@ -305,8 +305,8 @@ float log_gamma(const T x) {
   }
 
   // Reflection formula
-  return LOG_PI - rc -
-      ::metal::log(::metal::abs(abs_x * ::metal::sinpi(abs_x)));
+  auto log_arg = abs_x * ::metal::abs(::metal::sinpi(abs_x));
```

Comment:
can we add a comment saying that we're workaround'ing a compiler problem here?
If feels like some sort of peephole optimization that shouldn't kick in, but it does.

Reply:
Added non-PyTorch reproducer to the PR description and submitted FB16415011